vectorstores/azureaisearch/azureaisearch.go

```diff
@@ -0,0 +1,171 @@
+package azureaisearch
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+// Store is a wrapper to use azure AI search rest API.
+type Store struct {
+	azureAISearchEndpoint string
+	azureAISearchAPIKey   string
+	embedder              embeddings.Embedder
+	client                *http.Client
+}
+
+var (
+	// ErrNumberOfVectorDoesNotMatch when providing documents,
+	// the number of vectors generated should be equal to the number of docs.
+	ErrNumberOfVectorDoesNotMatch = errors.New(
+		"number of vectors from embedder does not match number of documents",
+	)
+	// ErrAssertingMetadata SearchScore is stored as float64.
+	ErrAssertingSearchScore = errors.New(
+		"couldn't assert @search.score to float64",
+	)
+	// ErrAssertingMetadata Metadata is stored as string.
+	ErrAssertingMetadata = errors.New(
+		"couldn't assert metadata to string",
+	)
+	// ErrAssertingContent Content is stored as string.
+	ErrAssertingContent = errors.New(
+		"couldn't assert content to string",
+	)
+)
+
+// New creates a vectorstore for azure AI search
+// and returns the `Store` object needed by the other accessors.
+func New(opts ...Option) (Store, error) {
+	s := Store{
+		client: http.DefaultClient,
+	}
+
+	if err := applyClientOptions(&s, opts...); err != nil {
+		return s, err
+	}
+
+	return s, nil
+}
+
+var _ vectorstores.VectorStore = Store{}
+
+// AddDocuments adds the text and metadata from the documents to the Chroma collection associated with 'Store'.
+// and returns the ids of the added documents.
+func (s Store) AddDocuments(
+	ctx context.Context,
+	docs []schema.Document,
+	options ...vectorstores.Option,
+) ([]string, error) {
+	opts := s.getOptions(options...)
+	ids := []string{}
+
+	texts := []string{}
+
+	for _, doc := range docs {
+		texts = append(texts, doc.PageContent)
+	}
+
+	vectors, err := s.embedder.EmbedDocuments(ctx, texts)
+	if err != nil {
+		return ids, err
+	}
+
+	if len(vectors) != len(docs) {
+		return ids, ErrNumberOfVectorDoesNotMatch
+	}
+	for i, doc := range docs {
+		id := uuid.NewString()
+		if err = s.UploadDocument(ctx, id, opts.NameSpace, doc.PageContent, vectors[i], doc.Metadata); err != nil {
+			return ids, err
+		}
+		ids = append(ids, id)
+	}
+
+	return ids, nil
+}
+
+// SimilaritySearch creates a vector embedding from the query using the embedder
+// and queries to find the most similar documents.
+func (s Store) SimilaritySearch(
```

Comment:
Can you make the receiver a pointer for consistency?

Reply:
sorry, can you be more specific?

Reply:
(s *Store) instead of (s Store)